### PR TITLE
upgrade cats effect to rc, replace unqiue with ce3 one

### PR DESCRIPTION
**Pull request by m-sp**

### CODE REVIEW

**Comment by rossabaker** [comment]
Argh.  I didn't get notifications on this, and just did this myself!

File: core/src/main/scala/org/typelevel/vault/Key.scala [hunk 1]
```diff
@@ -30,22 +31,22 @@ import cats.implicits._
...
+final class Key[A] private (private[vault] val unique: Unique.Token) {
   override def hashCode(): Int = unique.hashCode()
 }
 
 object Key {
   /**
    * Create A Typed Key
    */
-  def newKey[F[_]: Sync, A]: F[Key[A]] = Unique.newUnique[F].map(new Key[A](_))
+  def newKey[F[_]: Sync, A]: F[Key[A]] = Sync[F].unique.map(new Key[A](_))
```
Comment (rossabaker) [inline core/src/main/scala/org/typelevel/vault/Key.scala:42]: All we need here is a `Unique` and a `Functor`.  The `Sync` constraint rules out `Spawn` unless we're also `Async`.  Maybe we should just take two constraints here?

**Review by rossabaker — COMMENTED** [review]
This looks about like what I did, and confuses me in the spot I was confused. :+1: when we resolve the double constraint question.

File: core/src/main/scala/org/typelevel/vault/Key.scala (hunk 1 above)
Comment (m-sp) [reply core/src/main/scala/org/typelevel/vault/Key.scala]: good point, https://github.com/typelevel/vault/pull/203/commits/afa2511a5559407cd2a59fc93b33f19ad2a1dd6e

**Review by rossabaker — APPROVED** [review]

**Pull request merged**